demos/python_demos/bert_question_answering_embedding_demo/bert_question_answering_embedding_demo.py

```diff
@@ -0,0 +1,360 @@
+
+    #small class to store context as text and tokens and its embedding vector
+    class ContextData:
+        def __init__(self, context, c_tokens_id, c_tokens_se):
+            self.context = context
+            self.c_tokens_id = c_tokens_id
+            self.c_tokens_se = c_tokens_se
+            self.c_emb = calc_emb(self.c_tokens_id, max_length_c)
+
+    paragraphs = get_paragraphs(args.input.split(','))
+    contexts_all = []
+
+    log.info("Indexing {} paragraphs...".format(len(paragraphs)))
+    for par in paragraphs:
+        c_tokens_id, c_tokens_se = text_to_tokens(par.lower(), vocab)
+        if not c_tokens_id:
+            continue
+
+        # get context as string and then encode it into token id list
+        # calculate number of tokens for context in each request.
+        # reserve 3 positions for special tokens
+        # [CLS] q_tokens [SEP] c_tokens [SEP]
+        if args.model_qa:
+            #to make context be able to pass model_qa together with question
+            c_wnd_len = max_length_qc - (max_length_q + 3)
+        else:
+            #to make context be able to pass model_emb without question
+            c_wnd_len =  max_length_c - 2
+
+        # token num between 2 neighbours context windows
+        # 1/2 means that context windows are interleaved by half
+        c_stride = c_wnd_len // 2
+
+        # init scan window
+        c_s, c_e = 0, min(c_wnd_len, len(c_tokens_id))
+
+        # iterate while context window is not empty
+        while c_e > c_s:
+            contexts_all.append(ContextData(par, c_tokens_id[c_s:c_e], c_tokens_se[c_s:c_e]))
+
+            # check that context window reach the end
+            if c_e == len(c_tokens_id):
+                break
+
+            # move to next window position
+            c_s, c_e = c_s+c_stride, c_e+c_stride
+
+            shift_left = max(0, c_e - len(c_tokens_id))
+            c_s, c_e = c_s -shift_left, c_e-shift_left
+            assert c_s >= 0, "start can be left of 0 only with window less than len but in this case we can not be here"
+
+    #loop to ask many questions
+    while True:
+        question = input('Type question (enter to exit):')
+        if not question:
+            break
+
+        log.info("---Stage 1---Calc question embedding and compare with {} context embeddings".format(len(contexts_all)))
+        q_tokens_id, _ = text_to_tokens(question.lower(), vocab)
+
+        q_emb = calc_emb(q_tokens_id, max_length_q)
+        distances = [(np.linalg.norm(c.c_emb - q_emb, 2), c) for c in contexts_all]
+        distances = sorted(distances, key=lambda x: x[0])
+        keep_num = min(args.best_n, len(distances))
+        distances_filtered = distances[:keep_num]
+
+        #print short list
+        log.info("The most closest contexts to question")
+        for i, (dist, c_data) in enumerate(distances_filtered):
+            log.info("{} embedding distance {} for context '{}'".format(i, dist, c_data.context))
```

Comment:
```suggestion
        print("The closest contexts to question:")
        for i, (dist, c_data) in enumerate(distances_filtered):
            print("#{}: embedding distance {} for context '{}'".format(i + 1, dist, c_data.context))
```

(this stuff should be printed rather than logged, since it's the actual output of the demo, not just some progress report)